### PR TITLE
Common SQLClient interface with implementation

### PR DESCRIPTION
**Pull request by EmadAlblueshi**

### CODE REVIEW

File: src/main/java/io/vertx/ext/asyncsql/AsyncSQLClient.java [hunk 1]
```diff
@@ -28,28 +26,6 @@
...
-  void close(Handler<AsyncResult<Void>> whenDone);
-
-  /**
-   * Returns a connection that can be used to perform SQL operations on. It's important to remember to close the
-   * connection when you are done, so it is returned to the pool.
-   *
-   * @param handler the handler which is called when the <code>JdbcConnection</code> object is ready for use.
-   */
-  void getConnection(Handler<AsyncResult<SQLConnection>> handler);
+public interface AsyncSQLClient extends SQLClient {
```
Comment (Narigo) [inline src/main/java/io/vertx/ext/asyncsql/AsyncSQLClient.java]: Is the `AsyncSQLClient` still in use somewhere then? Might be in use in client code, I suppose?

Comment (EmadAlblueshi) [inline src/main/java/io/vertx/ext/asyncsql/AsyncSQLClient.java]: As far as I can remember yes

**Comment by EmadAlblueshi** [comment]
@Narigo The CI fails probably because `SQLClient` interface in `vertx-sql-common` not yet merged.

**Comment by pmlopes** [comment]
Yes we need to merge the 3 PR s common, jdbc, async

File: src/main/java/io/vertx/ext/asyncsql/AsyncSQLClient.java (hunk 1 above)
Comment (EmadAlblueshi) [inline src/main/java/io/vertx/ext/asyncsql/AsyncSQLClient.java]: @Narigo I have reviewed what you said I think I can get rid of `AsyncSQLClient` 

@pmlopes WDYT ?

**Comment by EmadAlblueshi** [comment]
:+1:

**Pull request merged**